### ISSUE TITLE
scikit-image 0.19 has changed behavior for values beyond the bounds of the original image

### ISSUE THREAD

**Issue by thedoctar** [sunpy/sunpy#5750]
### Description

When I run sunpy.self_test, it fails due to an assertion failure as the relative difference is infinity.

For more info, see log: [log.txt](https://github.com/sunpy/sunpy/files/7664867/log.txt)

### Expected vs Actual behavior

When the expected value is zero, assert should not fail when the result is close to zero.

### Steps to Reproduce

```python
import sunpy
sunpy.self_test()
```

### System Details
```
==============================
sunpy Installation Information
==============================

General
#######
OS: Ubuntu (20.04, Linux 5.4.0-91-generic)
Arch: 64bit, (x86_64)
sunpy: 3.1.2
Installation path: /home/user/.local/lib/python3.8/site-packages

Required Dependencies
#####################
packaging: 21.3
astropy: 5.0
numpy: 1.21.4
parfive: 1.5.0

Optional Dependencies
#####################
asdf: 2.8.1
beautifulsoup4: 4.10.0
cdflib: 0.3.20
dask: 2021.11.2
drms: 0.6.2
glymur: 0.9.6
h5netcdf: 0.11.0
h5py: 3.6.0
matplotlib: 3.5.0
mpl-animators: 1.0.0
pandas: 1.3.4
python-dateutil: 2.8.2

reproject: 0.8
scikit-image: 0.19.0
scipy: 1.7.3
sqlalchemy: 1.4.27
tqdm: 4.62.3
zeep: 4.1.0
```

**Comment by ayshih** [comment]
This is due to a change in behavior in `scikit-image` 0.19.0 (see https://github.com/sunpy/sunpy/pull/5742#issuecomment-986009890), and the tests pass on `scikit-image` 0.18.3.  Until we figure out how we're going to address this in SunPy, we have temporarily pinned the dependency requirement to <0.19.0 (#5743).

**Comment by thedoctar** [comment]
Hi Ayshih, it seems to me that the failures still are rather mild. We're talking about a result of order 1e-3 when it should be zero. That seems to me like a numerical integrator returning a number compatible with zero.

**Comment by thedoctar** [comment]
Maybe here is a hypothesis as to why this is the case. The skimage transformation uses some interpolation. Maybe this interpolation sets the boundaries of the image to a decaying small number when shrinking. I.e. the resulting image has a boundary that smoothly goes to zero, instead of sharply in expected array.

**Comment by ayshih** [comment]
The issue is that when we use the keyword argument `constant` to specify what value to use beyond the bounds of the original image, `scikit-image` 0.19 now passes that on to `scipy` as the keyword argument `grid-constant` instead of `constant`, which uses the constant value on a fixed grid, but additionally applies interpolation between the grid points.  For non-trivial orders of interpolation, this results in a new "ringing" behavior with `scikit-image` 0.19.  There needs to be a discussion on how to deal with this change in behavior.

**Comment by ayshih** [comment]
Closed by #5867: scikit-image is no longer the rotation implementation used by default
